src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1627,18 +1629,14 @@ EnsureRelationCanBeDistributed(Oid relationId, Var *distributionColumn,
 
 	ErrorIfTableIsACatalogTable(relation);
 
-	/* verify target relation does not use identity columns */
-	if (RelationUsesIdentityColumns(relationDesc))
-	{
-		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("cannot distribute relation: %s", relationName),
-						errdetail("Distributed relations must not use GENERATED "
-								  "... AS IDENTITY.")));
-	}
 
-	/* verify target relation is not distributed by a generated columns */
+	/* verify target relation is not distributed by a generated columns
+	 * except generated identity
+	 */
 	if (distributionMethod != DISTRIBUTE_BY_NONE &&
-		DistributionColumnUsesGeneratedStoredColumn(relationDesc, distributionColumn))
+		DistributionColumnUsesGeneratedStoredColumn(relationDesc, distributionColumn) &&
+		!DistributionColumnUsesGeneratedIdentityColumn(relationDesc, distributionColumn)
```

Comment:
`attributeForm->attgenerated` has only one value, so if its value is `ATTRIBUTE_GENERATED_STORED` , `DistributionColumnUsesGeneratedIdentityColumn(relationDesc, distributionColumn)` will return false. Meaning, you don't need the third condition here.